modules/dnn/src/layers/scale_layer.cpp

```diff
@@ -291,5 +291,131 @@ Ptr<Layer> ShiftLayer::create(const LayerParams& params)
+        inputs_arr.getMatVector(inputs);
+        CV_Assert(inputs.size() == 1);
+        CV_Assert(!mean_per_pixel && blobs.size() == 3 || blobs.size() >= 2);
+        ++num_iter;
+    }
+
+    void forward(InputArrayOfArrays inputs_arr, OutputArrayOfArrays outputs_arr, OutputArrayOfArrays internals_arr) CV_OVERRIDE
+    {
+        CV_TRACE_FUNCTION();
+        CV_TRACE_ARG_VALUE(name, "name", name.c_str());
+
+        std::vector<Mat> inputs, outputs;
+        inputs_arr.getMatVector(inputs);
+        outputs_arr.getMatVector(outputs);
+
+        CV_Assert_N(outputs.size() == 1, blobs.size() == 3, inputs.size() == 1);
+
+        Mat &inpBlob = inputs[0];
+        Mat &outBlob = outputs[0];
+
+        float* inpData = (float*)inpBlob.data;
+        float* outData = (float*)outBlob.data;
+
+        Mat data_mean_cpu = blobs[1];
+        Mat data_mean_per_channel_cpu = blobs[2];
+
+        const int numWeights = blobs[1].total();
+        CV_Assert(numWeights != 0);
+
+        if (num_iter <= recompute_mean) {
+            data_mean_cpu *= (num_iter - 1);
+            const int batch = inputs[0].size[0];
+            float alpha = 1.0 / batch;
+
+            for (int i = 0; i < batch; ++i)
+            {
+                Mat inpSlice(1, numWeights, CV_32F, inpData);
+                inpSlice = alpha * inpSlice;
+
+                add(data_mean_cpu.reshape(1, 1), inpSlice, data_mean_cpu.reshape(1, 1));
+                inpData += numWeights;
+            }
+            data_mean_cpu *= (1.0 / num_iter);
+
+            std::vector<int> newsize = {3, data_mean_cpu.total(2)};
+            reduce(data_mean_cpu.reshape(1, newsize), data_mean_per_channel_cpu, 1, REDUCE_SUM, CV_32F);
+
+            int area = blobs[1].total(2);
+            data_mean_per_channel_cpu *= (1.0 / area);
+        }
+
+        MatShape inpShape = shape(inpBlob);
+
+        inpData = (float*)inpBlob.data;
+        if (mean_per_pixel) {
+            int numSlices = inputs[0].size[0];
+            for (int i = 0; i < numSlices; ++i)
+            {
+                Mat inpSlice(1, numWeights, CV_32F, inpData);
+                Mat outSlice(1, numWeights, CV_32F, outData);
+
+                add(inpSlice, (-1) * data_mean_cpu, outSlice);
+                inpData += numWeights;
+                outData += numWeights;
+            }
+        } else {
+            int numSlices = inpShape[1];
+            int count = numWeights / numSlices;
+
+            for (int i = 0; i < numSlices; ++i)
+            {
+                Mat inpSlice(1, count, CV_32F, inpData);
+                Mat outSlice(1, count, CV_32F, outData);
+                float coeff = data_mean_per_channel_cpu.reshape(1, 1).at<float>(0, i);
+                outSlice = inpSlice - coeff;
+
+                inpData += count;
+                outData += count;
+            }
+        }
+    }
+
+    virtual bool supportBackend(int backendId) CV_OVERRIDE
+    {
+        return backendId == DNN_BACKEND_OPENCV;
+    }
```

Comment:
I think we don't need to override that now:
https://github.com/opencv/opencv/blob/3d14dd4e3953aa7b7e2ab3a388ca7bf08fe99d99/modules/dnn/src/dnn.cpp#L4340-L4343